dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java

```diff
@@ -857,6 +860,21 @@ public Connection getConnectionFromDatasource(DataSource ds, KerbUsage useKerb,
             } catch (ResourceException ex) {
                 throw AdapterUtil.toSQLException(ex);
             }
+            
+            //Workaround for Oracle Driver + IBM JDK issue with non-connection builder path
+            if (mcf.dsConfig.get().sendGSSCredentialOnOracleBuilder) {
+                try {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
+                        Tr.debug(this, tc, "Using Connection Builder path for Kerberos");
+                    Object oracleConnectionBuilder = ds.getClass().getMethod("createConnectionBuilder").invoke(ds);
+                    Object oracleConnectionBuilder2 = oracleConnectionBuilder.getClass().getMethod("gssCredential", GSSCredential.class).invoke(oracleConnectionBuilder, gssCredential);
+                    Method m = oracleConnectionBuilder.getClass().getMethod("build");
+                    m.setAccessible(true);
```

Comment:
Why would this need setAccessible(true)?  Whether it is Oracle vendor-specific API or JDBC 4.3 spec API, the method we are invoking ought to be public and should not require this.